src/utils.cpp

```diff
@@ -764,5 +781,241 @@ bool executePreAction(const nlohmann::json& i_parsedConfigJson,
 
     return true;
 }
+
+int updateHardware(const types::Path& i_hwPath, const types::VpdData& i_data,
+                   const nlohmann::json& i_jsonObj)
+{
+    int rc = -1;
+    try
+    {
+        // Get the VPD type and perform write operation
+        auto l_vpdStartOffset = utils::getVPDOffset(i_jsonObj, i_hwPath);
+
+        // Read the VPD data into a vector.
+        types::BinaryVector l_vpdVector;
+        utils::getVpdDataInVector(i_hwPath, l_vpdVector, l_vpdStartOffset);
+
+        // This will detect the type of parser required.
+        std::shared_ptr<vpd::ParserInterface> parser =
+            ParserFactory::getParser(l_vpdVector, i_hwPath, l_vpdStartOffset);
+
+        rc = parser->write(i_hwPath, i_data);
+    }
+    catch (const std::exception& e)
+    {
+        throw e;
+    }
+
+    return rc;
+}
+
+int updateDbus(const types::Path& i_invPath, const types::VpdData& i_data)
```

Comment:
As we discussed, am not seeing function pointer for read and write calls to particular 'DBus service' is being made .